libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -1267,7 +1269,7 @@ - (SFSDKOAuthClient *)fetchOAuthClient:(SFOAuthCredentials *)credentials cached:
         __weak typeof(self) weakSelf = self;
         client = [SFSDKOAuthClient clientWithCredentials:credentials configBlock:^(SFSDKOAuthClientConfig *config) {
             __strong typeof(self) strongSelf = weakSelf;
-            config.loginHost = strongSelf.loginHost;
+            config.loginHost = credentials.domain ?: strongSelf.loginHost;
```

Comment:
This is food for thought down the road, not necessarily this PR.  It strikes me that creating `SFOAuthCredentials` instances to have some of their required fields populated non-atomically is fundamentally problematic, and something of an anti-pattern.  We do a number of checks like this in credentials objects, when we shouldn't have to: `SFOAuthCredentials` should have a number of fields that are required for it to be a valid data object, which in turn would imply requiring non-null values in initializers. `SFOAuthCredentials` should not exist without a `protocol`, `domain`, `identifier`, `clientId`, and `redirectUri`.

Something to think about for the future. I suppose you could debate whether you take that even further and say that a valid SFOAuthCredentials object implies a user that has authenticated at some point, so further values like `instanceUrl` become required too.  But regardless of that, the fields above should never not be a part of the credentials object—you can't do anything credentials-related with that object, without them.

Reply:
@khawkins Thanks for the feedback. You've made some good arguments. In addition we should be thinking of the mutability of this object and its implications. Will add it to future improvements.

Reply:
@sgoldberg-sfdc That's essentially it, though changing the implementation of `SFOAuthCredentials` could end up changing the shape of a lot of the current code that utilizes it—so who can be sure what this particular branch of code would look like, post-refactor.  But yes: if you say that credentials always have the appropriate bootstrapping data elements—and as @trooper2013 says, potential considerations on their immutability—in theory you should not need this check, as you'll always have the login host value that's relevant to this code, within the credentials object.

Reply:
in terms of mutability, wouldn't mutability of this property be necessary to support org splits/instance refresh?

Reply:
Depends on whether you considered the current credentials valid in the event that such data needed to change.  You may forego mutability in favor of cloning with updated data elements. It just depends on your feeling for the sanctity of the data.

Given that login host information should only change rarely, if ever, for a given user, I'd consider its mutability to be more of a liability than an asset, in general practice.